File: src/python/function_vs_method.py
# Συνάρτηση
def greet(name):
    return f"Hello, {name}!"


class Greeter:
    def __init__(self, name):
        self.name = name

    # Μέθοδος
    def greet(self):
        return f"Hello, {self.name}!"


print(greet("Alice"))  # Έξοδος: Hello, Alice!
g = Greeter("Bob")
print(g.greet())  # Έξοδος: Hello, Bob!

File: src/cpp/function_vs_method.cpp
#include <iostream>
#include <string>

// Συνάρτηση
std::string greet(const std::string& name) { return "Hello, " + name + "!"; }

class Greeter {
 private:
  std::string name;

 public:
  Greeter(const std::string& name) : name(name) {}

  // Μέθοδος
  std::string greet() const { return "Hello, " + name + "!"; }
};

int main() {
  std::cout << greet("Alice") << std::endl;  // Έξοδος: Hello, Alice!
  Greeter g("Bob");
  std::cout << g.greet() << std::endl;  // Έξοδος: Hello, Bob!
  return 0;
}
